### PR TITLE
Feat/48 cache2

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

Use a Hash of selected values in the AdmTile rather than the JSONized string.

## Issue(s)

Issue #48

### CODE REVIEW

File: src/server/cache.rs [hunk 1]
```diff
@@ -86,25 +88,35 @@ async fn tile_cache_updater(state: &ServerState) {
         match result {
             Ok(response) => {
                 //trace!("tile_cache_updater: {:#?}", response);
-                let tiles = match serde_json::to_string(&response) {
-                    Ok(tiles) => tiles,
+                let json_tiles = match serde_json::to_string(&response) {
```
Comment (pjenvey) [inline src/server/cache.rs:91]: Until we know more about how large this cache will eventually be, in retrospect, storing solely the json in it is really a premature optimization on my part.

What do you think about just storing the `AdmTileResponse` struct and having it derive `Eq/PartialEq` (`AdmTile` needs them too). We could store the rendered json along side it too (maybe that's even premature before benchmarks)  -- but I'd rather we stick to comparing the cache values with strict equality (via `AdmTileResponse`) for now vs relying on a hash code (and potential collisions).

Maybe we'll need something like this later if the cache is too large but I'm leaning towards keeping this dead simple until we need something different.

File: src/adm.rs [hunk 2]
```diff
@@ -252,6 +257,18 @@ pub struct AdmTile {
     pub position: Option<u8>,
 }
 
+impl AdmTile {
```
Comment (pjenvey) [inline src/adm.rs:260]: FYI - this doesn't include `id` (or `position` but that's not very relevant) -- `#[derive(Hash)]` would have included them all and gave you the hash impl for free (only difference is the caller needs to create a DefaultHasher itself to pass to every `hash` method call)

File: src/server/location.rs [hunk 3]
```diff
@@ -32,11 +40,13 @@ impl From<&RequestHead> for LocationResult {
         if let Some(loc_string) = headers.get(GOOG_LOC_HEADER) {
             dbg!("Found google header", loc_string);
             let mut parts: Vec<&str> = loc_string.to_str().unwrap_or("").split(',').collect();
-            return Self {
-                city: parts.pop().map(ToOwned::to_owned),
-                country: parts.pop().map(ToOwned::to_owned),
+            let reply = Self {
+                country: parts.pop().map(|v| v.trim().to_owned()),
```
Comment (pjenvey) [inline src/server/location.rs:44]: Also note the changes here I made in #78 --

- don't want city (yet), just country and subdivision
- it's gonna be "--custom-request-header 'X-Client-Geo-Location: {client_region},{client_region_subdivision}"
- `client_region` is mostly country. `client_region_subdivision` is a "Unicode CLDR subdivision ID" which includes a 2 letter country prefix we strip

File: src/server/cache.rs (hunk 1 above)
Comment (jrconlin) [reply src/server/cache.rs:91]: I'd be ok with using the `Eq/PartialEq` trait. I guess I was more thinking that we want to compare elements of the tile. (e.g. we may not need to compare `id` or `position` for instance.) This is kinda per the other comments. Honestly, i was more figuring that there could be variations in the returned tile values that might churn the cache and this way we'd have a normalization point.

File: src/server/location.rs (hunk 3 above)
Comment (jrconlin) [reply src/server/location.rs:44]: I'll correct the header we get location from.

**Comment by jrconlin** [comment]
closing for reasons

**Pull request closed without merging**